h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -785,8 +805,11 @@ public final double likelihood(double yr, double ym) {
         case gamma:
           if (yr == 0) return -2;
           return -2 * (Math.log(yr / ym) - (yr - ym) / ym);
-        case tweedie:
-          return deviance(yr, ym); //fixme: not really correct, not sure what the likelihood is right now
+        case tweedie: // we ignore the a(y,phi,p) term in the likelihood calculation here since we are not optimizing over them
+          double temp = 0;
+          temp += _var_power==2?Math.log(ym):Math.pow(ym,_twoMinusVarPower)*_oneOtwoMinusVarPower;
+          temp -= yr*(_var_power==1?Math.log(ym):Math.pow(ym,_oneMinusVarPower)*_oneOoneMinusVarPower);
+          return temp; // ignored the a(y,phi,p) term as it is a constant for us
         default:
```

Comment:
I am calculating the deviance according to this:

<img width="424" alt="image" src="https://user-images.githubusercontent.com/7231712/58065986-6f798580-7b3c-11e9-85e6-976bfb9a5ff0.png">